helm/krr-enforcer/templates/enforcer-cert-job.yaml

```diff
@@ -0,0 +1,191 @@
+              openssl req -x509 -new -nodes -key ca.key -subj "/CN=robusta-krr-enforcer-ca" -days 365 -out ca.crt
+
+              # Generate a server key and certificate signing request (CSR)
+              echo "Generating server certificate..."
+              SERVICE_NAME={{ .Release.Name }}-krr-enforcer
+              NAMESPACE={{ .Release.Namespace }}
+              DNS_NAME=${SERVICE_NAME}.${NAMESPACE}.svc
+              
+              openssl genrsa -out server.key 2048
+              cat > server.conf <<EOF
+              [req]
+              req_extensions = v3_req
+              distinguished_name = req_distinguished_name
+              [req_distinguished_name]
+              [v3_req]
+              basicConstraints = CA:FALSE
+              keyUsage = nonRepudiation, digitalSignature, keyEncipherment
+              extendedKeyUsage = serverAuth
+              subjectAltName = @alt_names
+              [alt_names]
+              DNS.1 = ${SERVICE_NAME}
+              DNS.2 = ${SERVICE_NAME}.${NAMESPACE}
+              DNS.3 = ${DNS_NAME}
+              DNS.4 = ${DNS_NAME}.cluster.local
+              EOF
+              
+              openssl req -new -key server.key -out server.csr -subj "/CN=${DNS_NAME}" -config server.conf
+
+              # Sign the server certificate with our CA
+              echo "Signing server certificate..."
+              cat > server-ext.conf <<EOF
+              [v3_req]
+              basicConstraints = CA:FALSE
+              keyUsage = nonRepudiation, digitalSignature, keyEncipherment
+              extendedKeyUsage = serverAuth
+              subjectAltName = @alt_names
+              [alt_names]
+              DNS.1 = ${SERVICE_NAME}
+              DNS.2 = ${SERVICE_NAME}.${NAMESPACE}
+              DNS.3 = ${DNS_NAME}
+              DNS.4 = ${DNS_NAME}.cluster.local
+              EOF
+              
+              openssl x509 -req -in server.csr -CA ca.crt -CAkey ca.key -CAcreateserial -out server.crt -days 365 -extensions v3_req -extfile server-ext.conf
+
+              # Create a Secret with the certificates
+              echo "Creating/updating secret..."
+              kubectl create secret tls ${SERVICE_NAME}-certs \
+                --cert=server.crt --key=server.key \
+                --dry-run=client -o yaml | kubectl apply -f -
+
+              # Update the secret to include the CA certificate
+              kubectl get secret ${SERVICE_NAME}-certs -o json | \
+                jq --arg ca "$(cat ca.crt | base64 | tr -d '\n')" \
+                  '.data["ca.crt"] = $ca' | \
+                kubectl apply -f -
+
+              # Get the CA bundle for the webhook
+              CA_BUNDLE=$(cat ca.crt | base64 | tr -d '\n')
+
+              # Set up the webhook configuration
+              echo "Creating/updating webhook configuration..."
+              cat <<EOF | kubectl apply -f -
+              apiVersion: admissionregistration.k8s.io/v1
+              kind: MutatingWebhookConfiguration
```

Comment:
I'm not sure I understand the flow here. You create the MutatingWebhookConfiguration here, but its also created in enforcer-webhook.yaml and they don't match exactly

Reply:
fixed, deleted the second webhook